Nos/Service/Relay/RelaySubscriptionManager.swift

```diff
@@ -179,6 +182,40 @@ actor RelaySubscriptionManagerActor: RelaySubscriptionManager {
         }
     }
     
+    /// Tracks that a relay sent us an AUTH message, so we can change the socket state to .authenticating
+    func trackAuthenticationRequest(from socket: WebSocket, responseID: RawNostrID) {
+        if let relayAddress = socket.url, let connection = socketConnections[relayAddress] {
+            // Close open subscriptions if any.
+            active.forEach { subscription in
+                if subscription.relayAddress == relayAddress {
+                    subscription.subscriptionStartDate = nil
+                    socket.write(string: "[\"CLOSE\", \"\(subscription.id)\"]")
+                }
+            }
+            connection.state = .authenticating(responseID)
+        }
+    }
+    
+    /// Checks the ID of an "OK" message from a relay to see if it matches any authentication events we have sent.
+    /// If it does we mark the relay as .connected. If it doesn't then we do nothing, so this function is safe to be 
+    /// called on every "OK" message. 
+    func checkAuthentication(success: Bool, from socket: WebSocket, eventID: RawNostrID, message: String?) -> Bool {
+        if let relayAddress = socket.url, let connection = socketConnections[relayAddress] {
```

Comment:
Could we flip this statement to a `guard` to avoid the three levels of `if` statements?